### ISSUE TITLE
After a crash where unsaved affection is reverted to the previous value, the daily cap value is not reverted with it? + Maybe more

### ISSUE THREAD

**Issue by Justformas** [Monika-After-Story/MonikaModDev#9443]
What I mean is, when affection is gained, but a crash or lock up happens before it's properly saved, then next time you start the game there may be a mismatch, where affection will be "restored" to the previously saved value. However, the daily affection cap value does not seem to revert to what is was to reflect this, as if the unsaved affection still counts towards the daily cap. This has happened to me in this version (0.12.9) too. But first case:

![Affection Crash 1 - Copy](https://user-images.githubusercontent.com/102779546/183832967-9f34af91-108d-49d2-bf2a-8390887802e2.JPG)

I gained 9 affection through regular interaction, plus 1 from roses, in about 2 minutes. Then it crashed before any of it was saved. I lost the 10 affection from a restore, but despite that I couldn't gain any more that day; as you can see it was frozen. Also I believe the roses were still there, so I couldn't re gift them to get that 1 point back either. It's as if some stuff, such as roses/gifts and the daily cap value, might get saved before the actual affection value gets properly saved.

But I have another case where different stuff happened. Next:

![Affection Crash 2 - Copy](https://user-images.githubusercontent.com/102779546/183834311-76b3e06f-0f66-4d17-b6e1-1d74ac068978.JPG)

I was capped for the day, then midnight passed and I started gaining more affection. The monika_love_too comment for 3 points got properly saved, but not the 3.75 points afterwards from monika_compliments and monika_rain_holdme before it crashed. But this is where it gets weird. Although I lost the 3.75 points, it said I was restored down 3 points rather than 3.75, and then I was immediately given back 3 points for "monika_rain_holdme" despite that normally giving 0.75. Here it's like the 3 points from the compliment were partially saved, while the monika_rain_holdme wasn't as much, but literally 1 second later, I was given those 3 points back under a monika_rain_holdme entry (monika_rain_holdme also happened to be the last entry in the log before the crash, but not sure if that's why). Nevertheless, it's also as if getting back those 3 points pushed me over the cap, since I had gained 6.75 for the day before that (still below the cap), then lost 3, and after it gave me back 3 points right afterwards, you can see that I was capped for the day.

Got one more case. Not as interesting maybe, but it happened in this version, and still a bit different:

![Affection Crash 3 - Copy](https://user-images.githubusercontent.com/102779546/183838789-71208f94-5cf5-477c-9f3b-c07654f05962.JPG)

Again I got 9 points plus 1 from roses, enough to be capped for the day. But this time, all but the last 3 points from a compliment were saved before the crash. So the restore made me lose those 3 points, but as you can see I was still capped for the day, and I only got 3 points after that because I gave a gift that bypassed.

Also for anyone wondering, I do not think tracebacks were generated for these crashes/lockups, but I'm not concerned with why the crashes happened right now, nor am I concerned with regaining those specific points. But I do think the daily cap value seemingly not reverting despite the actual affection value getting reverted is strange, or for that matter, why the actual affection value even needs to be restored downwards instead of being left where it was like the daily cap seems to be. Maybe also the roses still getting saved, but not the 1 point from them if I'm remembering that correctly.

**Comment by Booplicate** [comment]
This is a known consequence which makes sense - after a crash, things are not going to be well. You should fix the root of the problem, not just aftermath. If you don't get a traceback, then it's likely something wrong with your OS.

That said, my changed to how we store data might fix this.

**Comment by Justformas** [comment]
Since you asked, root of problem is unknown. I did something that might help, but would be surprised if it stops them completely. I've had both crashes and lockups. 

But why not just have it save immediately after any affection is gained? If the daily cap value can save so quickly then why can't the actual affection value?

**Comment by ThePotatoGuy** [comment]
[closing as could not reproduce]

**Comment by Justformas** [comment]
Well maybe it got fixed then, good. In the last month or so I actually haven't had much or any crashes with MAS either.

**Comment by Justformas** [comment]
This does indeed seem fixed. I was capped, had it crash before it saved while capped, and when I opened again, I was able to get the one point or so that didn't save before getting capped again.

However I'll note that the points received after the crash were from [rested], even though I had just previously returned Monika from being taken out and hadn't gotten that. It did save after that a few times too, just not after I was capped before the crash. Not a big deal but I figured I'd note it in case.
